api/src/main/java/org/openmrs/api/db/hibernate/NativeIfNotAssignedIdentityGenerator.java

```diff
@@ -33,20 +37,15 @@
  * 
  * @author paul.shemansky@gmail.com
  */
-public class NativeIfNotAssignedIdentityGenerator extends IdentityGenerator implements Configurable {
+public class NativeIfNotAssignedIdentityGenerator extends IdentityGenerator implements IdentifierGenerator, Configurable {
 	
 	private String entityName;
 	
 	@Override
-	public Serializable generate(SharedSessionContractImplementor session, Object entity) throws HibernateException {
-		Serializable id;
+	public Object generate(SharedSessionContractImplementor session, Object entity) throws HibernateException {
 		EntityPersister persister = session.getEntityPersister(entityName, entity);
 		// Determine if an ID has been assigned.
-		id = persister.getIdentifier(entity, session);
-		if (id == null) {
```

Comment:
Did you intentionally remove the calls to super.generate() when id is null?

Reply:
Yes this was intentional, there was no longer a super method that can be called